### PR TITLE
Lesson info display

### PR DESCRIPTION
**Pull request by AngelaLuo49021**
## Status:
:rocket: Ready

## Description :star2:

Restyling of the Lesson Page + filtering
Lets just get this merged in before anisha sees!
 
Screen Shot 2020-04-24 at 2 57 12 PM
Screen Shot 2020-04-24 at 2 57 18 PM
Screen Shot 2020-04-24 at 2 57 25 PM
Screen Shot 2020-04-24 at 2 57 35 PM

### CODE REVIEW

File: src/components/LessonDateDisplay/LessonDateDisplay.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import "./LessonDateDisplay.scss";
 
 function LessonDateDisplay(props) {
+  const { firebase } = props;
```
Comment (chyku) [inline src/components/LessonDateDisplay/LessonDateDisplay.js]: instead of manually assinging firebase as props you can do `function LessonDateDisplay({ firebase })`

File: src/components/LessonInfoDisplay/LessonInfoDisplay.js [hunk 2]
```diff
@@ -1,13 +1,40 @@
 import React, { useState } from "react";
+import { Col, Row } from "reactstrap";
 import { compose } from "recompose";
 import { withRouter, Redirect } from "react-router-dom";
+import { useRef } from "react";
+import useOutsideClick from "../WordSelector/useOutsideClick";
 import "./LessonInfoDisplay.scss";
 
-function LessonInfoDisplay(props) {
+const LessonInfoDisplay = props => {
```
Comment (chyku) [inline src/components/LessonInfoDisplay/LessonInfoDisplay.js]: why did you make this a const function with this syntax, what difference is this making?

File: src/components/LessonInfoDisplay/LessonInfoDisplay.js [hunk 3]
```diff
@@ -22,42 +49,49 @@ function LessonInfoDisplay(props) {
...
+          <div className="info-display">
+            <div className="column">
+              <div className="word-group-display">
+                <div className="lesson-group-name">{props.template}</div>
+              </div>
+              <div className="word-display">
+                <div className="grey-box">
+                  <div className="word-group-name">
+                    {" "}
+                    {props.lesson.wordGroup}{" "}
```
Comment (chyku) [inline src/components/LessonInfoDisplay/LessonInfoDisplay.js]: i'm pretty sure prettier did this but what would happen if you removed the `{" "}`s

File: src/components/LessonInfoDisplay/LessonInfoDisplay.js [hunk 4]
```diff
@@ -22,42 +49,49 @@ function LessonInfoDisplay(props) {
...
+                </div>
+                <div className="words-list">
+                  {props.lesson.words.map(word => (
+                    <div className="words">{word}</div>
+                  ))}
+                </div>
+              </div>
+            </div>
+
+            <Col className="fat-column">
```
Comment (chyku) [inline src/components/LessonInfoDisplay/LessonInfoDisplay.js]: this class name

File: src/components/LessonInfoDisplay/LessonInfoDisplay.scss [hunk 5]
```diff
@@ -1,95 +1,156 @@
 @import "../../pages/Index/index.scss";
 
-.LessonInfo {
+.lesson-info {
```
Comment (chyku) [inline src/components/LessonInfoDisplay/LessonInfoDisplay.scss:3]: ty for changing the class names :))

File: src/components/LessonInfoDisplay/LessonInfoDisplay.scss [hunk 6]
```diff
@@ -1,95 +1,156 @@
...
+  overflow: hidden;
+  margin-left: 1.8em;
+}
+.grey-box, .purple-box {
   height: 3vw;
-  border: 1px solid black;
-  border-top-left-radius: 8px;
-  border-top-right-radius: 8px;
-  background-color: lightgray;
+  background-color: #C4C4C4;
```
Comment (chyku) [inline src/components/LessonInfoDisplay/LessonInfoDisplay.scss:69]: what color is this and does it need a variable? Ik gene made some variables for light gray in his branch though

File: src/components/LessonInfoDisplay/LessonInfoDisplay.scss [hunk 7]
```diff
@@ -1,95 +1,156 @@
...
+.fat-column  {
+  width: 60vw
 }
 
-.ButtonContainer {
-  display: flex;
-  flex-direction: row;
-  justify-content: flex-start;
-  align-items: flex-start;
+.margins {
```
Comment (chyku) [inline src/components/LessonInfoDisplay/LessonInfoDisplay.scss]: can u make this class name more specific

File: src/components/LessonNameDisplay/LessonNameDisplay.scss [hunk 8]
```diff
@@ -1,19 +1,32 @@
 @import "../../pages/Index/index.scss";
 
 .NameIcon {
```
Comment (chyku) [inline src/components/LessonNameDisplay/LessonNameDisplay.scss:3]: Would b great if you could update these names too but u dont have to

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js [hunk 9]
```diff
@@ -1,67 +1,86 @@
...
 import "./CustomLessonsDisplay.scss";
 import { compose } from "recompose";
+import { Col, Row } from "reactstrap";
 import { withRouter, Redirect, useHistory } from "react-router-dom";
 import LessonDateDisplay from "../../components/LessonDateDisplay/LessonDateDisplay";
 import LessonInfoDisplay from "../../components/LessonInfoDisplay/LessonInfoDisplay";
 import LessonNameDisplay from "../../components/LessonNameDisplay/LessonNameDisplay";
 
 const CustomLessonsDisplay = props => {
   const { firebase } = props;
```
Comment (chyku) [inline src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: can you update this to be `function CustomLessonsDisplay({ firebase }) {` and then delete the declaration of firebase

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js [hunk 10]
```diff
@@ -1,67 +1,86 @@
...
-  const [adminLessons, setAdminLessons] = useState([]);
+  const [showLessons, setShowLessons] = useState([]);
+  const [allLessons, setAllLessons] = useState([]);
+  const [filterType, setFilterType] = useState();
+  const [filterGroup, setFilterGroup] = useState();
```
Comment (chyku) [inline src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: could be cool if you grouped filter-related states together with a comment on top

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js [hunk 11]
```diff
@@ -1,67 +1,86 @@
...
-      } else {
-        return a.dueDate["seconds"] - b.dueDate["seconds"];
-      }
-    });
-    setAdminLessons(sortedLessons);
-  }
+  /**
+   *
+   * wait can teachers delete whole lessons that they've already made?????
+   */
```
Comment (chyku) [inline src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: actually we do not have a delete feature atm i'm pretty sure

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js [hunk 12]
```diff
@@ -1,67 +1,86 @@
...
     setTimeout(() => {
       // TODO: figure out better solution to resolve this
       // Get custom lessons made by admin
-      firebase.getAdminCustomLessons(ADMIN_ACCOUNT).then(lesson => {
-        setAdminLessons(lesson);
+      firebase.getAdminCustomLessons(ADMIN_ACCOUNT).then(lessons => {
+        setAllLessons(lessons);
       });
-    }, 50); // Timeout for firebase to update and display updated lessons properly
+    }, 60); // Timeout for firebase to update and display updated lessons properly
```
Comment (chyku) [inline src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: why did you change the timeout LOL

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js [hunk 13]
```diff
@@ -1,67 +1,86 @@
...
+        .then(deploymentAccount => {
+          deploymentNameMap[id] = deploymentAccount.username;
+        })
+        .catch(error => console.error("Error getting custom lesson: ", error));
+    });
+
+    //Wait for deploymentNameMap to finish resolving
+    while (Object.keys(deploymentNameMap).length == 0) {
+      await new Promise(r => setTimeout(r, 500));
+    }
```
Comment (chyku) [inline src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: lam fixed this jank code actually so it would be great if you took it from him again

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js [hunk 14]
```diff
@@ -70,53 +89,99 @@ const CustomLessonsDisplay = props => {
...
+                    className="drop-down"
+                    onClick={() => setFilterType(key)}
+                  >
+                    {TEMPLATE_LESSON_MAP[key]}
+                  </Dropdown.Item>
+                ))}
+              </div>
+            </DropdownButton>
+          </Col>
+          {/* Cant filter by wordgroups if phonics is selected */}
```
Comment (chyku) [inline src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: i am not a fan of this comment in the middle of the jsx

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js [hunk 15]
```diff
@@ -70,53 +89,99 @@ const CustomLessonsDisplay = props => {
...
+          <Row className="justify-content-md-center">
+            {allLessons
+              .filter(
+                lesson =>
+                  (!filterType || lesson.lessonTemplate == filterType) &&
+                  (!filterGroup ||
+                    lesson.wordGroup == TEMPLATE_WORD_GROUPS[filterGroup])
+              )
+              .map(lesson => (
+                // <div key={lesson.id} onClick={() => handleClick(lesson)}>
```
Comment (chyku) [inline src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: this isn't good version control :((

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js [hunk 16]
```diff
@@ -70,53 +89,99 @@ const CustomLessonsDisplay = props => {
...
+                <Col key={lesson.id} onClick={() => handleClick(lesson)}>
+                  <LessonNameDisplay lessonName={lesson.lessonName} />
+                </Col>
+                // </div>
```
Comment (chyku) [inline src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: same here LOL

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.scss [hunk 17]
```diff
@@ -1,26 +1,64 @@
...
+  text-transform: lowercase;
   font-family: $assignments-roboto-font;
-  width: 100vw;
-  display: flex;
-  flex-wrap: wrap;
-  flex-direction: row;
-  position: absolute;
+  font-size: 1em;
+  &hover {
+    background-color: "#f5f5f5";
```
Comment (chyku) [inline src/pages/CustomLessonsDisplay/CustomLessonsDisplay.scss]: why is this in quotes

**Review by chyku — CHANGES_REQUESTED** [review]
good :clap: stuff :clap:

File: src/components/LessonInfoDisplay/LessonInfoDisplay.js (hunk 4 above)
Comment (AngelaLuo49021) [reply src/components/LessonInfoDisplay/LessonInfoDisplay.js]: but its a fat column

Comment (AngelaLuo49021) [reply src/components/LessonInfoDisplay/LessonInfoDisplay.js]: thicc column?

File: src/components/LessonDateDisplay/LessonDateDisplay.js (hunk 1 above)
Comment (AngelaLuo49021) [reply src/components/LessonDateDisplay/LessonDateDisplay.js]: ope dis component not even supposed to be here

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js (hunk 9 above)
Comment (chyku) [reply src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: hello

File: src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js (hunk 11 above)
Comment (chyku) [reply src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js]: can u delete this comment block

**Review by chyku — CHANGES_REQUESTED** [review]
pls dont ignore

**Review by chyku — APPROVED** [review]
